### DISCUSSION TITLE
from __future__ import annotations breaks typer in lower python version

### DISCUSSION THREAD

**Discussion by Asthestarsfalll** [fastapi/typer#977] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# test.py
from __future__ import annotations

import typer
from typer import Argument as CArg
from typer import Option as  COp
from typing_extensions import Annotated

app = typer.Typer(rich_markup_mode="rich")


@app.command()
def foo(
    entry: Annotated[str, CArg()] = "test",
) -> None:
    print('foo')

app()
```

### Description

In python 3.8 and 3.9, run 
```
python test.py foo
```
It will raise error:
```
Usage: test.py [OPTIONS]
Try 'test.py --help' for help.
╭─ Error ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Got unexpected extra argument (foo)                                                                                        │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```
Python 3.10 works well.

### Operating System

Linux

### Typer Version

0.12.5

### Python Version

3.8/3.9

**Comment by ckutlu** [comment]
I also ran into the same issue on python 3.9, when trying to use a custom class with a parser:

```
app = typer.Typer()
@app.command()
def subcommand(foo: Annotated[Foo, typer.Argument(parser=Foo.from_str)]): ...
```
results in

```
RuntimeError: Type not yet supported: <class '__main__.Foo'>
```

This was for a helper python script where I was using python 3.9 simply because the library that the helper script is "helping" has to support lower versions. I can easily switch to a recent version in this case, but it took me a while to figure out that the issue was a bug and not my misuse of typer.

@tiangolo I'd be happy to make a PR with a disclaimer in the documentation if you could point me the right place to do this. If you think the fix is easy, I could also take a jab at it as well.

**Comment by kratsg** [comment]
Identical:
- https://github.com/fastapi/typer/discussions/802
